app/lib/features/onboarding/pages/upload_avatar_page.dart

```diff
@@ -167,19 +165,14 @@ class UploadAvatarPage extends ConsumerWidget {
     );
   }
 
-  Widget _buildSkipActionButton(BuildContext context) {
+  Widget _buildSkipActionButton(BuildContext context, WidgetRef ref) {
     return OutlinedButton(
       key: UploadAvatarPage.skipBtn,
       onPressed: () async {
-        if (context.mounted) {
-          await handleNotificationPermission(context);
-        }
-        if (context.mounted) {
-          await handleCalendarPermission(context);
-        }
-        if (context.mounted) {
-          context.goNamed(Routes.analyticsOptIn.name);
-        }
+        if (!context.mounted) return;
+        await ref.read(postLoginSignupProvider).initialize(context);
```

Comment:
I like the centralisation into a service. that's good. But I think also the final `goNamed` should just be dealt with by that thing - and rather than it being a service I think it should be a `startupWizzard`-page with its own route, so we could also just simply route there from here or elsewhere.

See comment on the service file.